request-manager/request/manager.go

```diff
@@ -56,6 +56,9 @@ type Manager interface {
 	// state from the proto.FinishRequest argument.
 	Finish(requestId string, finishParams proto.FinishRequest) error
 
+	// Fail a pending request (if it can't be started for some reason).
+	Fail(requestId string) error
```

Comment:
Hmm, since this only works on requests that are pending, how about calling this `FailPending` to make that extra clear?